drivers: sensor: bmi160: Fix device definition on bus I2C

vinceWii4 opened the pull request:
When the bmi160 device is located on bus i2c:
 - the anymotion trigger can't be mapped to int-pin
 - interrupt status register can't be read

Signed-off-by: Vincent De Carvalho <vincent.decarvalho@wiifor.com>

vinceWii4 committed:
drivers: sensor: bmi160: Fix device definition on bus I2C

Fix following issues when the bmi160 device is located on bus i2c:
- the anymotion trigger can't be mapped to int-pin
- interrupt status register can't be read

Signed-off-by: Vincent De Carvalho <vincent.decarvalho@wiifor.com>

sjg20 reviewed (approved):
Not keen on the #if stuff but I don't have a better idea

vinceWii4 committed:
drivers: sensor: bmi160: Define DT_DRV_COMPAT in bmi160_trigger.c

Fix a missing compilation directive in bmi160_trigger.c which cause bad definition in bmi160.h included file.

Signed-off-by: Vincent De Carvalho <vincent.decarvalho@wiifor.com>

vinceWii4 commented:
1. I updated _**drivers: sensor: bmi160: Fix device definition on bus I2C**_ commit message body
2. I added a commit to fix an issue in bmi160_trigger.c

MaureenHelm commented on drivers/sensor/bmi160/bmi160_trigger.c:
```diff
@@ -5,6 +5,8 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+#define DT_DRV_COMPAT bosch_bmi160
```
Please squash this change into the previous commit

galak commented:
@vinceWii4 Any updates.

galak commented:
This also needs to be submitted to master branch first before we backport.
